File: PullViewOC/PullViewOC/RelatedFile/DragView.m
//
//  DragView.m
//  PullViewOC
//
//  Created by travelsky on 2018/4/10.
//  Copyright © 2018年 wangdicen. All rights reserved.
//

#import "DragView.h"

@implementation DragView
{
    CGPoint center;
}

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        self.userInteractionEnabled = YES;
        UIPanGestureRecognizer *pan = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(panHandle:)];
        self.gestureRecognizers = @[pan];
    }
    return self;
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    [self.superview bringSubviewToFront:self];
    center = self.center;
}

- (void)panHandle:(UIPanGestureRecognizer *)gesture
{
    CGPoint translation = [gesture translationInView:self.superview];
    self.center = CGPointMake(center.x + translation.x, center.y + translation.y);
}
/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: PullViewSwift/PullViewSwift/RelatedFile/DragView.swift
//
//  DragView.swift
//  PullViewSwift
//
//  Created by travelsky on 2018/4/11.
//  Copyright © 2018年 wangdicen. All rights reserved.
//

import Foundation
import UIKit

class DragView: UIView {
    var originalCenter: CGPoint;
    
    override init(frame: CGRect) {
        self.originalCenter = CGPoint.zero;
        super.init(frame: frame);
        self.isUserInteractionEnabled = true;
        let pan = UIPanGestureRecognizer.init(target: self, action: #selector(DragView.panAction(gesture:)));
        self.addGestureRecognizer(pan);
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        self.superview?.bringSubview(toFront: self);
        originalCenter = self.center;
    }
    
    @objc func panAction(gesture:UIPanGestureRecognizer){
        let translation = gesture.translation(in: self.superview);
        self.center = CGPoint.init(x: originalCenter.x + translation.x, y: originalCenter.y + translation.y);
        
    }
}


